Makefile

```
@@ -1,44 +1,47 @@
-EMACS_INIT  := /root/.emacs.d/init.el
-EMACS_LOAD  := -l $(EMACS_INIT)
-EMACS_FLAGS := --batch --kill $(EMACS_LOAD)
-ORG_FILES   := $(wildcard *.org)
-HTML_FILES  := $(patsubst %.org, %.html, $(ORG_FILES))
-DOCKER_IMG  := commonlispbr/emacs
-USER        := $(shell id -u):$(shell id -g)
-DOCKER_RUN  := docker run  -w /tmp \
-	                       -v $(shell pwd):/tmp \
-                           --rm \
-                           -t
-STATUS_PREFIX    := "\033[1;32m[+]\033[0m "
-ATTENTION_PREFIX := "\033[1;36m[!]\033[0m "
+EMACS_INIT          := /root/.emacs.d/init.el
+EMACS_LOAD          := -l $(EMACS_INIT)
+EMACS_FLAGS         := --batch --kill $(EMACS_LOAD)
+ORG_FILES           := $(wildcard *.org)
+HTML_FILES          := $(patsubst %.org, %.html, $(ORG_FILES))
+STATIC_FILES_FOLDER := static/
+DOCKER_IMG          := commonlispbr/emacs
+MINIFY_IMG          := tdewolff/minify
+USER                := $(shell id -u):$(shell id -g)
+DOCKER_RUN          := docker run  -w /tmp \
+	                               -v $(shell pwd):/tmp \
+                                   --rm \
+                                   -t
+
+STATUS_PREFIX       := "\033[1;32m[+]\033[0m "
 
 .PHONY: clean shell
 
-all: $(HTML_FILES) copy
+all: setup $(HTML_FILES) minify
 
 shell:
 	$(DOCKER_RUN) -i --entrypoint=/bin/zsh $(DOCKER_IMG)
 
+setup:
+	@printf $(STATUS_PREFIX); echo "CREATING STATIC FOLDER: $(STATIC_FILES_FOLDER)"
+	@mkdir -p static
+
 %.html: %.org
 	@printf $(STATUS_PREFIX); echo "COMPILING: $< -> $*.html"
 	@$(DOCKER_RUN) $(DOCKER_IMG) $< $(EMACS_FLAGS) -f org-html-export-to-html
-	@$(DOCKER_RUN) --entrypoint=/bin/chown $(DOCKER_IMG) $(USER) "$*.html"
+	@mv $*.html $(STATIC_FILES_FOLDER)/$*.html
+	@printf $(STATUS_PREFIX); echo "MINIFYING: $*.html"
+	@$(DOCKER_RUN) --entrypoint="" $(MINIFY_IMG) sh -c 'minify $(STATIC_FILES_FOLDER)/$*.html -o $(STATIC_FILES_FOLDER)/$*.html'
 
-copy:
-	@mkdir -p static
-	@find . \
-	      "(" -name "*.html" -o -name "*.css" -o -name "*.js" ")" \
-	      -not -path "./static/*" \
-	      -not -path "./.github/*" \
-	      -exec cp {} static \;
-	@find . \
-	      -name "*.html" \
-	      -not -path "./static/*" \
-	      -not -path "./.github/*" \
-	      -exec mv {} static \;
+minify:
+	@printf $(STATUS_PREFIX); echo "MINIFYING AND BUNDLING CSS FILES"
+	@$(DOCKER_RUN) --entrypoint="" $(MINIFY_IMG) sh -c 'minify --bundle org-theme/dist/long/bundle.css css/syntax.css -o $(STATIC_FILES_FOLDER)/long.css'
```

Why this empty entrypoint is necessary? Which is the default one?

---

I read the Dockerfile recipe from this image, they set a init.sh script which wraps minify binary.

Maybe a less convoluted option:

```suggestion
	@$(DOCKER_RUN) --entrypoint="/usr/bin/minify" $(MINIFY_IMG) --bundle org-theme/dist/long/bundle.css css/syntax.css -o $(STATIC_FILES_FOLDER)/long.css
```

or even:

```suggestion
	@$(DOCKER_RUN) $(MINIFY_IMG) --bundle org-theme/dist/long/bundle.css css/syntax.css -o $(STATIC_FILES_FOLDER)/long.css
```

---

I tried not passing an entry point, as suggested by the documentation. For some reason, I got a permission error with `init.sh`.

I didn't manage to solve it, but by calling `minify` directly the error didn't happen.

If you know another way to solve this permission error so we don't have to pass an entry point, please let me know.

---

I made a change based on your first suggestion.